tableExport.js

```diff
@@ -32,9 +32,34 @@ THE SOFTWARE.*/
 						pdfLeftMargin:20,
 						escape:'true',
 						htmlContent:'false',
-						consoleLog:'false'
+						consoleLog:'false',
+						name: 'file' // the name that will be shown to the user in the dialog.
 				};
-                
+                		
+                		// This is a function which opens up a 
+                		// download file dialog using an invisible
+                		// <a> tag. It solves the problem of long
+                		// and unusual names. So instead of using
+                		// windows.open(uri), we'll now use this 
+                		// cleaner way.
+                		function openDownloadFileDialog(name,uri) {
```

Comment:
This is just a helper function, which opens the window and assigns the file a name. Its the core of this change.